fabric-common/types/index.d.ts

```diff
@@ -61,7 +61,7 @@ export interface ICryptoSuite {
 	hash(msg: string, opts: any): string;
 	importKey(pem: string, opts?: KeyOpts): ICryptoKey | Promise<ICryptoKey>;
 	setCryptoKeyStore(cryptoKeyStore: ICryptoKeyStore): void;
-	sign(key: ICryptoKey, digest: Buffer): Buffer;
+	sign(key: ICryptoKey, digest: Buffer): Buffer | Promise<Buffer>;
```

Comment:
if it is promisified, then no need to preseve type `Buffer`

Reply:
Current cryptosuite of `Default X509` and `HSM X509` return a simple `Buffer` and I wanted to keep this PR non-breaking. But again this interface is never used inside the `sdk` , it won't affect even if I remove `Buffer` from return type.

Question : should I remove `Buffer` from return type ?

Reply:
If we want to cherry-pick the change back to _release-2.2_, I think we should keep it a non-breaking change. If it's going only into _main_ then I'm comfortable with a breaking change.

Reply:
@bestbeforetoday thank you for reviewing this PR , I will change it to return only `Promise<Buffer>`.

Reply:
The knock-on changes of return type for `IdentityContext.sign()` and `ServiceAction.sign()` from `Buffer` to `Promise<Buffer>` will also be breaking changes. So it's looking harder to make this a non-breaking change